substrate/client/network/src/request_responses.rs

```diff
@@ -62,14 +62,48 @@ use std::{
 	sync::Arc,
 	task::{Context, Poll},
 	time::{Duration, Instant},
+	fmt::{Display,Formatter},
+	fmt,
 };
 
 pub use libp2p::request_response::{Config, InboundFailure, OutboundFailure, RequestId};
 
+/// Adding a custom OutBoundFailure, not depending on libp2p
+#[derive(Debug, thiserror::Error)]
+#[allow(missing_docs)]
```

Comment:
If you use doc comments `/// dial-failure`, may be you don't need to implement the `Display` trait manually. If this doesn't work, you can use the `#[error("dial-failure")]` annotation.